sky/provision/aws/instance.py

```diff
@@ -152,14 +156,17 @@ def cleanup_ports(
 ) -> None:
     """See sky/provision/__init__.py"""
     assert provider_config is not None, (cluster_name, provider_config)
+    if 'ports' not in provider_config:
+        return
     region = provider_config['region']
     ec2 = aws.resource(
         'ec2',
         region_name=region,
         config=config.Config(retries={'max_attempts': BOTO_MAX_RETRIES}))
     # TODO(tian): Add a function to generate SG name for AWS, then replace here
     # and backend_utils::write_cluster_config
-    sg_name = f'sky-sg-{common_utils.user_and_hostname_hash()}-{cluster_name}'
+    sg_name = (f'sky-sg-{common_utils.user_and_hostname_hash()}'
+               f'-{common_utils.hash_cluster_name(cluster_name)}')
```

Comment:
In fact, the `common_utils.hash_cluster_name` function already implements this: it will check cluster name length and truncate and append a hash if too long, otherwise we just return the original cluster name. The function name is a little bit confusing 🤧 I'll consider other name

Reply:
Now renamed to `truncate_and_hash_cluster_name`. If you have another better name, please let me know 👀